### ISSUE TITLE
upgrade to freeform 5.0.0 Error: Attribute class "Solspace\Freeform\Attributes\Property\Input\Textarea" not found

### ISSUE THREAD

**Issue by bob-pixeldeluxe** [solspace/craft3-freeform#978]
Unable to upgrade to freeform 5.0.0

**Steps to reproduce**
1. Upgrade to 5.0.0 from: 4.1.13
2. Error follows with database changes.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 4.5.14
- Freeform Version: 5.0.0
- Freeform Edition: lite
- Fresh Install or Upgrade: upgrade

**Additional context**

**Comment by bob-pixeldeluxe** [comment]
Full error: 
`
One of Freeform’s migrations failed.

Error: Attribute class "Solspace\Freeform\Attributes\Property\Input\Textarea" not found

Migration: Solspace\Freeform\migrations\m230101_100010_FF4to5_MigrateForms

Output:

> create index idx_dzhimcxlxosjrlagmsxxpsqfyhnwlcqtmdbr on {{%freeform_forms}} (order) ... done (time: 0.014s)
Exception: Attribute class "Solspace\Freeform\Attributes\Property\Input\Textarea" not found (/var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Library/Helpers/AttributeHelper.php:19)
#0 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Library/Helpers/AttributeHelper.php(19): ReflectionAttribute->newInstance()
#1 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Bundles/Attributes/Property/PropertyProvider.php(98): Solspace\Freeform\Library\Helpers\AttributeHelper::findAttribute(Object(ReflectionProperty), 'Solspace\\Freefo...')
#2 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Bundles/Attributes/Property/PropertyProvider.php(51): Solspace\Freeform\Bundles\Attributes\Property\PropertyProvider->getEditableProperties(Object(Solspace\Freeform\Form\Settings\Implementations\GeneralSettings))
#3 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/migrations/m230101_100010_FF4to5_MigrateForms.php(69): Solspace\Freeform\Bundles\Attributes\Property\PropertyProvider->setObjectProperties(Object(Solspace\Freeform\Form\Settings\Implementations\GeneralSettings), Array)
#4 /var/www/html/vendor/craftcms/cms/src/db/Migration.php(49): Solspace\Freeform\migrations\m230101_100010_FF4to5_MigrateForms->safeUp()
#5 /var/www/html/vendor/craftcms/cms/src/db/MigrationManager.php(233): craft\db\Migration->up(true)
#6 /var/www/html/vendor/craftcms/cms/src/db/MigrationManager.php(149): craft\db\MigrationManager->migrateUp(Object(Solspace\Freeform\migrations\m230101_100010_FF4to5_MigrateForms))
#7 /var/www/html/vendor/craftcms/cms/src/services/Updates.php(252): craft\db\MigrationManager->up()
#8 /var/www/html/vendor/craftcms/cms/src/controllers/BaseUpdaterController.php(493): craft\services\Updates->runMigrations(Array)
#9 /var/www/html/vendor/craftcms/cms/src/controllers/UpdaterController.php(203): craft\controllers\BaseUpdaterController->runMigrations(Array, 'restore-db')
#10 [internal function]: craft\controllers\UpdaterController->actionMigrate()
#11 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#12 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#13 /var/www/html/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('migrate', Array)
#14 /var/www/html/vendor/craftcms/cms/src/web/Application.php(305): yii\base\Module->runAction('updater/migrate', Array)
#15 /var/www/html/vendor/craftcms/cms/src/web/Application.php(693): craft\web\Application->runAction('updater/migrate')
#16 /var/www/html/vendor/craftcms/cms/src/web/Application.php(262): craft\web\Application->_processUpdateLogic(Object(craft\web\Request))
#17 /var/www/html/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#18 /var/www/html/public_html/index.php(12): yii\base\Application->run()
#19 {main}`

**Comment by gustavs-gutmanis** [comment]
Hi @bob-pixeldeluxe 

Would you be able to try out a fix for this as we cannot reproduce it?
In your `composer.json` set the Freeform package's version to `dev-fix/SFT-840`

```json
{
  "require": {
    ...
    "solspace/craft-freeform": "dev-fix/SFT-840",
  },
  ...
}
```

**Comment by gustavs-gutmanis** [comment]
Also could you take a look in the `vendor/solspace/craft-freeform/packages/plugin/src/Attributes/Property/Input` folder and send me a screenshot of all the files that are in there, please?

**Comment by bob-pixeldeluxe** [comment]
Hi, using dev-fix/SFT-840 worked for us.

![Screenshot 2024-01-23 at 11 54 57](https://github.com/solspace/craft-freeform/assets/50947605/21a08bc7-58c4-4e72-9e57-e56d34ecbed2)

**Comment by gustavs-gutmanis** [comment]
Thank you for the fast response, I'll prepare this fix for the next release.

**Comment by kjmartens** [comment]
This has now been fixed in Freeform 5.0.1. 🙂
